src/ws.rs

```diff
@@ -255,7 +254,8 @@ async fn handle_messages(client_ws_rcv: &mut SplitStream<WebSocket>,
                 break;
             }
         };
-        handle_message(&game_id, &client_id, msg, &games, word_generator.clone()).await;
+        let word_generator = word_generator.clone();
+        handle_message(&game_id, &client_id, msg, &games, word_generator).await;
```

Comment:
I honestly don't quite understand why this is necessary. If you don't make this change, I think the relevant error is:

```
258 |         handle_message(&game_id, &client_id, msg, &games, word_generator.clone()).await;
    |                                                           --------------         ^^^^^^ await occurs here, with `word_generator` maybe used later
    |                                                           |
    |                                                           has type `&impl WordGenerator` which is not `Send`
```

What stuck out to me here is this part:

> has type `&impl WordGenerator` which is not `Send`

Rust seems to be trying to pass a reference (given away by the leading `&`) instead of moving the value itself, which would be just `impl WordGenerator`. I'm guessing this is some weird type inference issue where the compiler just doesn't get what you're trying to do.

By separating the clone onto an extra line, the correct type is inferred for `word_generator`, which we then pass to `handle_message` with the inferred type, so the problem goes away (that is my hypothesis, at least :D)